src/array/utf8/mutable.rs

```diff
@@ -368,10 +381,6 @@ impl<O: Offset> MutableUtf8Array<O> {
             self.validity.as_mut().unwrap(),
             iterator,
         );
-
```

Comment:
This extend might be called a lot, which means we have quadratic behavior on the `null_count` compute.